pkg/prompts/prompts.go

```diff
@@ -129,16 +148,58 @@ func NoBlankStringValidator(s string) error {
 	return nil
 }
 
+// Validator for App name
+func appNameValidator(name string) error {
+	if name == "" {
+		return fmt.Errorf("application name cannot be empty")
+	}
+
+	if !unicode.IsLetter(rune(name[0])) && !unicode.IsDigit(rune(name[0])) {
+		return fmt.Errorf("application name must start with a letter or digit")
+	}
+
+	if name[len(name)-1] == '-' || name[len(name)-1] == '_' || name[len(name)-1] == '.' {
+		return fmt.Errorf("application name must end with a letter or digit")
+	}
+
+	for _, r := range name {
+		if !unicode.IsLetter(r) && !unicode.IsDigit(r) && r != '-' && r != '_' && r != '.' {
+			return fmt.Errorf("application name can only contain letters, digits, '-', '_', and '.'")
+		}
+	}
+
+	if len(name) > 63 {
+		return fmt.Errorf("application name cannot be longer than 63 characters")
+	}
+
+	return nil
+}
+
 // RunDefaultableStringPrompt runs a prompt for a string variable, returning the user string input for the prompt
 func RunDefaultableStringPrompt(customPrompt config.BuilderVar, defaultValue string, validate func(string) error, Stdin io.ReadCloser, Stdout io.WriteCloser) (string, error) {
-	var validatorFunc func(string) error
 	if validate == nil {
-		validatorFunc = NoBlankStringValidator
+		validate = NoBlankStringValidator
+	}
+
+	validatorFunc := func(input string) error {
+		// Allow blank inputs because defaults are set later
+		if input == "" {
+			return nil
+		}
+		if customPrompt.Name == "APPNAME" {
```

Comment:
instead of including this into every prompt call can we put this one layer higher in the call stack? we already have a `validate` function as an arg, so we can just pass it for that

Reply:
ah okay i see that this would require some additional refactoring for the validator to work